src/Exceptions/Handler.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+namespace Vinelab\Bowler\Exceptions;
+
+use PhpAmqpLib\Exception\AMQPProtocolChannelException;
+use PhpAmqpLib\Exception\AMQPProtocolConnectionException;
+use Vinelab\Bowler\Contracts\BowlerExceptionHandler as ExceptionHandler;
+
+/**
+ * @author Kinane Domloje <kinane@vinelab.com>
+ */
+class Handler
+{
+    /**
+     * The BowlerExceptionHandler contract bound app's exception handler.
+     */
+    private $exceptionHandler;
+
+    public function __construct(ExceptionHandler $handler)
+    {
+        $this->exceptionHandler = $handler;
+    }
+
+    /**
+     * Map php-mqplib exceptions to Bowler's.
+     *
+     * @param \Exception $e
+     * @param array      $parameters
+     * @param array      $arguments
+     *
+     * @return mix
+     */
+    public function handleServerException(\Exception $e, $parameters = [], $arguments = [])
+    {
+        if ($e instanceof AMQPProtocolChannelException) {
+            $e = new DeclarationMismatchException($e->getMessage(), $e->getCode(), $e->getFile(), $e->getLine(), $e->getTrace(), $e->getPrevious(), $e->getTraceAsString(), $parameters,  $arguments);
```

Comment:
It should be sufficient to instantiate with `new DeclarationMismatchException($e)` instead of having to extract the data from the previous exception and manually pass them to the next one.